modules/compiler/spirv-ll/source/builder_core.cpp

```diff
@@ -3010,7 +3010,11 @@ cargo::optional<Error> Builder::create<OpPtrAccessChain>(
       basePointeeTy, base, indexes, module.getName(op->IdResult()),
       IRBuilder.GetInsertBlock());
 
-  if (basePointeeTy->isStructTy()) {
+  // Check member decorations if the input indexes field is not empty
+  // as in this case we are accessing an element.
+  // We do this by checking the size of indexes vector is greater than one as
+  // the element field is always added at the start.
+  if (basePointeeTy->isStructTy() && indexes.size() > 1) {
     checkMemberDecorations(basePointeeTy, indexes, op->IdResult());
```

Comment:
If we're going to enforce that `checkMemberDecorations` should only be called when `indexes.size() > 1` then it makes sense to enforce that at all call sites, and `assert` in `checkMemberDecorations` itself. Because this can happen also in `OpInBoundsAccessChain` and others.

So maybe just make `checkMemberDecorations` early exit if there aren't enough indices, and then we fix all 4 calls to it at once?

Reply:
That's a fair point, I've done that.